File: 283. Move Zeroes/Move Zeroes.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
	temp = 0
    
    nums.each_with_index{|n,i| n == 0 ? temp += 1: nums[i-temp] = n}

    (nums.size - temp).upto(nums.size-1){|i| nums[i] = 0}
end

File: 283. Move Zeroes/Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    var temp = 0;
    
    nums.forEach(function(num,i){
        num === 0 ? temp+=1 : nums[i - temp] = num;
    })
    
    for(var i = nums.length - temp; i < nums.length; i++) {
        nums[i] = 0
    }
};
